ENH: Show colors in terminology selector

cpinter opened the pull request:
The recommended colors for are shown as a color square for each type in the terminology navigator widget.

![image](https://user-images.githubusercontent.com/1325980/106263622-841c8500-621c-11eb-9589-d1106a41f631.png)

lassoan commented on Modules/Loadable/Terminologies/Widgets/qSlicerTerminologyNavigatorWidget.cxx:
```diff
@@ -1170,49 +1170,62 @@ void qSlicerTerminologyNavigatorWidget::populateTypeTable()
+  int noneTypeExists = 0;
+  QTableWidgetItem* noneItem = nullptr;
+  if (searchTerm.empty())
+  {
```
indentation

lassoan reviewed (approved):
This looks great, thank you!

cpinter commented:
I fixed the indentation and added a check not to show color if the type did not have a color to begin with, like this
![image](https://user-images.githubusercontent.com/1325980/106296012-091d9380-6249-11eb-84c6-1730dca37947.png)

lassoan commented:
When does it happen that the type does not have a color? When it has a modifier? In that case, it would be better to show the first available color (usually they are the same or very similar for all modifiers).

cpinter commented:
Yes it happens when they have modifiers. I think I can get the first modifier. Not sure how much it will slow things down, probably not noticeably.

cpinter commented:
Done, the first modifier's color is used if found.

lassoan reviewed (changes requested):
This will be good, there is just a build error on linux according to CircleCI.

cpinter commented:
I fixed the error.

lassoan commented:
Thank you!
